app/services/concerns/teachers/switch_lead_provider_helper.rb

```diff
@@ -0,0 +1,117 @@
+module Teachers
+  module SwitchLeadProviderHelper
```

Comment:
I wasn't really sure what to name this.  I don't think it should use the word Helper, but I could do with some input on appropriate naming.